pkg/osquery/tables/xfconf/xfconf.go

```diff
@@ -0,0 +1,230 @@
+//go:build linux
+// +build linux
+
+package xfconf
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/launcher/pkg/dataflatten"
+	"github.com/kolide/launcher/pkg/osquery/tables/dataflattentable"
+	"github.com/kolide/launcher/pkg/osquery/tables/tablehelpers"
+	"github.com/osquery/osquery-go/plugin/table"
+	"golang.org/x/exp/maps"
+)
+
+// Provides configuration settings for devices using XFCE desktop environment.
+// See https://docs.xfce.org/xfce/xfce4-session/advanced#files_and_environment_variables
+// for a list of the files and environment variables we check for configuration.
+
+var xfconfChannelXmlPath string = filepath.Join("xfce4", "xfconf", "xfce-perchannel-xml")
+
+type xfconfTable struct {
+	logger log.Logger
+}
+
+func TablePlugin(logger log.Logger) *table.Plugin {
+	t := &xfconfTable{
+		logger: logger,
+	}
+
+	return table.NewPlugin("kolide_xfconf", dataflattentable.Columns(table.TextColumn("username")), t.generate)
```

Comment:
Do you think it's worth returning the `path` we found the data at? I think it might be.

Reply:
That's a good idea, let me add it 🙂

Reply:
All set -- examples in the PR description updated also